modules/core/src/test/java/org/apache/ignite/internal/client/thin/TimeoutTest.java

```diff
@@ -217,4 +217,120 @@ public void testClientTimeoutOnOperation() throws Exception {
             }
         }
     }
+
+    /**
+     * Test that connection timeout is independent of request timeout during connection establishment.
+     */
+    @Test
+    @SuppressWarnings("ThrowableNotThrown")
+    public void testConnectionTimeoutIndependentOfRequest() throws Exception {
+        ServerSocket sock = new ServerSocket();
+        sock.bind(new InetSocketAddress("127.0.0.1", DFLT_PORT));
+
+        CountDownLatch connectionAccepted = new CountDownLatch(1);
+
+        IgniteInternalFuture<?> fut = GridTestUtils.runAsync(() -> {
+            try {
+                Socket accepted = sock.accept();
+                connectionAccepted.countDown();
+
+                Thread.sleep(2000);
+
+                U.closeQuiet(accepted);
+            }
+            catch (Exception e) {
+                throw new IgniteException("Accept thread failed: " + e.getMessage(), e);
+            }
+        });
+
+        long startTime = System.currentTimeMillis();
+
+        try {
+            ClientConfiguration cfg = new ClientConfiguration()
+                .setAddresses("127.0.0.1:" + DFLT_PORT)
+                .setConnectionTimeout(500)
+                .setRequestTimeout(10000);
+
+            GridTestUtils.assertThrowsWithCause(
+                () -> Ignition.startClient(cfg),
+                ClientConnectionException.class
+            );
+        }
+        finally {
+            U.closeQuiet(sock);
+        }
+
+        long elapsed = System.currentTimeMillis() - startTime;
+
+        assertTrue("Should fail around connection timeout (500ms), not request timeout",
+            elapsed >= 450 && elapsed < 2000);
+
+        assertTrue("Connection should have been accepted", connectionAccepted.await(1, TimeUnit.SECONDS));
+
+        fut.get();
+    }
+
+    /**
+     * Test that request timeout is independent of connection timeout during operations.
+     */
+    @Test
+    @SuppressWarnings("ThrowableNotThrown")
+    public void testRequestTimeoutIndependentOfConnection() throws Exception {
```

Comment:
Similar for this test:
1. Set server and connection timeous to Integer.MAX_VALUE
2. Just check that exception contains timeout exception (X.hasCause(IgniteFutureTimeoutCheckedException.class))